### PR TITLE
 Unit Testing

### PR DESCRIPTION
**Pull request by bernabasy**
RUBY / DATABASES AND INTERVIEWING: [OOP school library: unit tests](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)
In this pull request, we have done the following:
- Add unit tests for all the methods in all the classes to:
   Book
   Person
   Student
   Teacher
   Classroom
   Rental
 - Add test for all classes that you created while implementing the Decorator design pattern.

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  end
+end
+
+# Test for author
+describe '#author' do
+  it 'should return the author' do
+    book = Book.new('SIMBA', 'Author-LION')
+    expect(book.author).to eq('Author-LION')
+  end
+end
```
Comment (danmainah) [inline spec/book_spec.rb:24]: - [x] @team great job on your tests, however, you forgot to add tests for the `add_rental` and `to_h` methods , its a project requirement to have unit tests for all methods in a class.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative '../person'
+
+describe Person do
+  context 'with one parameters' do
+    it 'takes three parameters' do
+      person = Person.new(age: 20, parent_permission: 'Y', name: 'john')
+      expect(person).to be_an_instance_of Person
+    end
+  end
```
Comment (danmainah) [inline spec/person_spec.rb:10]: - [x] @team great job on your tests, however, you forgot to add tests for the `add_rental`, `correct_name`, `can_use_services?`, `of_age?` and `to_h` methods, its a project requirement to have unit tests for all methods in a class.

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    expect(student).to be_an_instance_of Student
+  end
+end
+
+describe '#play_hooky' do
+  it 'should show play_hooky symbol' do
+    student = Student.new(classroom: 6, age: 13, parent_permission: 'Yes', name: 'Mark')
+    expect(student.play_hooky).to eq '¯\(ツ)/¯'
+  end
+end
```
Comment (danmainah) [inline spec/student_spec.rb:15]: - [x]  @team great job on your tests, however, you forgot to add tests for `classroom=` method, it's a project requirement to have unit tests for all methods in a class.

File: spec/rental_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    it 'should add the rental to the book and person rentals' do
+      book = { 'title' => 'The Great Gatsby', 'author' => 'F. Scott Fitzgerald', 'rentals' => [] }
+      person = { 'name' => 'John Doe', 'age' => 25, 'parent_permission' => true, 'rentals' => [] }
+      date = '2022-01-01'
+      rental = Rental.new(date, person, book)
+
+      expect(book['rentals'].last).to eq rental
+      expect(person['rentals'].last).to eq rental
+    end
+  end
```
Comment (danmainah) [inline spec/rental_spec.rb:27]: - [x] @team great job on your tests, however, you forgot to add a test for the  `to_h` method, its a project requirement to have unit tests for all methods in a class.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @bernabasy and @felixDev22 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
- [x] @bernabasy and @felixDev22 kindly ensure your issues page link in the reader file is working correctly currently it's broken
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello Team!👋 

## Things to Appreciate 
- [ ] Correct Flow ✅ 
- [ ] Clean Code 👍 
- [ ] No Linter Errors 🔥 
- [ ] Descriptive Pull Request and Summary 👌 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**